test/unit/engine/apply-changeset/tasks/create-add-entities-task.test.ts

```diff
@@ -39,7 +39,23 @@ describe('createAddEntitiesTask', () => {
       },
     }
 
-    context.client.cma.entries.create = sinon.stub().resolves(createdEntry)
+    class Spy {
```

Comment:
Just curious why you chose to write it like this (with an ad-hoc class definition). Nothing wrong with that, just that I haven't seen that in our tests anywhere so far.

Reply:
oh yeah, I was fighting a bit with sinon to make it work like both a spy and a stub and that was not really working out for some reason, so it was just faster to write a similar thing, I made it a class to be stateful and record number of calls